docs/source/blueprints/BP005.rst

```diff
@@ -0,0 +1,291 @@
+  * Maintainer Group members (this will be used to block merges)
+* Supported Platforms
+* Supported execution method (Local vs Remote)
+  We intend to flag modules that have the capability of acting on remote
+  systems. This will probably be a broad classification, just for informational
+  purposes because unfortunately we have way too many ways of executing code
+  remotely.
+
+  We don't hope to address a complete standardization of the remote execution
+  APIs just yet, but we want to at least bring some visibility to it.
+
+For instance, the textual category and description will allow users to search
+for modules based on their interest.  Also, our pipelines will parse that data
+field containing the list of supported platforms, and will trigger tests on
+those platforms. Users will then be able to see the results of those tests on
+each of the platforms defined, giving them quality assurance and coverage
+there.
+
+The exact format of the metadata file is still to be defined, but a possibility
+is to define a JSON schema that will be provided on this new repository.
+
+Maintainability
+===============
+
+As described above, each library must follow the new standard and keep that
+same level for future versions. Because of that, new PRs must be approved by at
+least one MAINTAINER and pass the basic checks pipelines that will be provided
+as part of the autils repository.
+
+Besides the automatic checks, each MAINTAINER is responsible for keeping the
+consistency of that library with the overall project.
+
+Release
+=======
+
+As a new and standalone repository, there will not be any type of release
+version sync between Avocado and autils.  Avocado may release, say, version
+100.0, while autils won’t have a release. Because of that, autils will be
+versioned at a different pace than Avocado.
+
+Every week an automatic job will be executed to determine if there is a
+potential need for a new release:
+
+* If the number of commits since last release has reached a threshold; AND
+* There is no open discussion with the tag “release-decision”:
+  * Creates a new discussion (with the "release-decision" tag) to run a poll
+    between all the MAINTAINERS, so they will have the opportunity to thumbs
+    up if there is a need for a new release;
+  * The link to the new discussion should be sent to all MAINTAINERS;
+  * If the decision is positive, a round-robin selected maintainer should be
+    responsible for coordinating the release it self;
+  * And the discussion should be closed
+* If there is any existing open discussion, the bot job should comment on the
+  same discussion with the updated list of commits and ping again the
+  MAINTAINERS, give them the opportunity to thumbs up based on the new
+  status;
+
+Manual release process could also be started if a MAINTAINER think it is a good
+timing (i.e: Security bug or critical bug it was fixed).
+
+Backwards Compatibility
```

Comment:
I am not sure whether it is proper to add one comment here, which is related utils API Compatibility.
In avocado-vt and tp-libvirt, those two repository heavily rely on some basic utils, such as process.run().
So it is critical to maintain those utils API Compatibility between different autils release, otherwise minor unCompatibility may lead to huge efforts change in avocado-vt and tp-libvirt.

Reply:
The Backwards Compatibility section is to describe what this proposal will "break" with the current code and how.

But indeed, I agree with you @chunfuwen. Each maintainer should be responsible to make sure to use the best practices (like deprecation warnings) while updating their versions.